framework/CodeInterfaces/Neutrino/neutrinoInterface.py

```diff
@@ -0,0 +1,219 @@
+"""
+Created on Jul 25, 2017
+
+@author: Emerald Ryan
+
+"""
+
+from __future__ import division, print_function, absolute_import
+import warnings
+warnings.simplefilter('default',DeprecationWarning)
+
+import os
+import lxml.etree as ET
+
+from CodeInterfaceBaseClass import CodeInterfaceBase
+
+class Neutrino(CodeInterfaceBase):
+  """
+    Provides code to interface RAVEN to Neutrino code
+    The name of this class represents the type in the RAVEN input file
+    e.g.
+    <Models>
+      <Code name="myName" subType="NeutrinoBase">
+      ...
+      </Code>
+      ...
+    </Models>
+
+  """
+
+  def generateCommand(self, inputFiles, executable, clargs=None,fargs=None,preExec=None):
+    """
+      See base class.  Collects all the clargs and the executable to produce the command-line call.
+      Returns tuple of commands and base file name for run.
+      Commands are a list of tuples, indicating parallel/serial and the execution command to use.
+      @ In, inputFiles, list, List of input files (length of the list depends on the number of inputs have
+            been added in the Step is running this code)
+      @ In, executable, string, executable name with absolute path (e.g. /home/path_to_executable/code.exe)
+      @ In, clargs, dict, optional, dictionary containing the command-line flags the user can specify in the input
+            (e.g. under the node < Code >< clargstype =0 input0arg =0 i0extension =0 .inp0/ >< /Code >)
+      @ In, fargs, dict, optional, a dictionary containing the axuiliary input file variables the user can specify
+            in the input (e.g. under the node < Code >< clargstype =0 input0arg =0 aux0extension =0 .aux0/ >< /Code >)
+      @ Out, returnCommand, tuple, tuple containing the generated command. returnCommand[0] is the command to
+            run the code (string), returnCommand[1] is the name of the output root
+    """
+    found = False
+
+    # Find the first file in the inputFiles that is an XML, which is what we need to work with.
+    for index, inputFile in enumerate(inputFiles):
+      if self._isValidInput(inputFile):
+        found = True
+        break
+    if not found:
+      raise Exception('No correct input file has been found. Got: '+' '.join(inputFiles))
+
+    #Determines the path to the input file
+    path = inputFiles[0].getAbsFile()
+
+    #Creates the output file that saves information that is outputted to the command prompt
+    #The output file name of the Neutrino results
+    outputfile = 'results'
+
+    #Creates run command tuple (['executionType','execution command'], output file root)
+    #The path to the Neutrino executable is specified in the RAVEN input file as the executable
+    # since it must change directories to run
+    returnCommand = [('serial','cd ' + executable + ' && Neutrino.exe --nogui --file ' + str(path) \
```

Comment:
this executable name is indicated in the XML input file under the node "executable".
It can not be hardcoded